specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/registries.json

```diff
@@ -0,0 +1,873 @@
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "SkuScaleType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Automatic",
+            "description": "Automatically scales node count."
+          },
+          {
+            "value": "Manual",
+            "description": "Node count scaled upon user request."
+          },
+          {
+            "value": "None",
+            "description": "Fixed set of nodes."
+          }
+        ]
+      },
+      "additionalProperties": false
+    },
+    "SkuSetting": {
+      "description": "SkuSetting fulfills the need for stripped down SKU info in ARM contract.",
+      "required": [
+        "name"
+      ],
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "[Required] The name of the SKU. Ex - P3. It is typically a letter+number code.",
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
+        },
+        "tier": {
+          "description": "This field is required to be implemented by the Resource Provider if the service has more than one tier, but is not required on a PUT.",
+          "default": "Basic",
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/SkuTier"
+        }
+      },
+      "additionalProperties": false
+    },
+    "StorageAccountDetails": {
+      "description": "Details of storage account to be used for the Registry",
+      "type": "object",
+      "properties": {
+        "systemCreatedStorageAccount": {
+          "$ref": "#/definitions/SystemCreatedStorageAccount",
+          "x-nullable": true
+        },
+        "userCreatedStorageAccount": {
+          "$ref": "#/definitions/UserCreatedStorageAccount",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "SystemCreatedAcrAccount": {
+      "type": "object",
+      "properties": {
+        "acrAccountSku": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "armResourceId": {
+          "description": "ARM ResourceId of a resource",
+          "$ref": "#/definitions/ArmResourceId",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "SystemCreatedStorageAccount": {
+      "type": "object",
+      "properties": {
+        "armResourceId": {
+          "description": "ARM ResourceId of a resource",
+          "$ref": "#/definitions/ArmResourceId",
+          "x-nullable": true
+        },
+        "storageAccountHnsEnabled": {
+          "type": "boolean"
+        },
+        "storageAccountType": {
+          "description": "Allowed values:\r\n\"Standard_LRS\",\r\n\"Standard_GRS\",\r\n\"Standard_RAGRS\",\r\n\"Standard_ZRS\",\r\n\"Standard_GZRS\",\r\n\"Standard_RAGZRS\",\r\n\"Premium_LRS\",\r\n\"Premium_ZRS\"",
+          "type": "string",
+          "x-nullable": true
+        },
+        "allowBlobPublicAccess": {
```

Comment:
use enum instead of booleans

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

Reply:
This would require us to make changes to Registry Management service code @fahdkmsft

Reply:
Not a blocking comment.